packages/ckeditor5-engine/src/view/domconverter.ts

```diff
@@ -732,6 +733,10 @@ export default class DomConverter {
 	 * @returns View selection.
 	 */
 	public domSelectionToView( domSelection: DomSelection ): ViewSelection {
+		if ( isGeckoRestrictedDomSelection( domSelection ) ) {
+			return new ViewSelection( [] );
+		}
+
```

Comment:
Please add tests for this case (tests that are already for the selection observer should also stay there).

Reply:
I'd add a link to the GH issue in code comment.